### ISSUE TITLE
Can't read loom files with scv.read>=0.1.4

### ISSUE THREAD

**Issue by juugii** [theislab/scvelo#20]
Hello,
I previously did some tries on scvelo 0.1.4.
Running scvelo 0.1.11.dev6+133abe9 after a git pull.
I can't read loom files anymore. I tried on my files and from downloaded examples:

```python
adata = scv.read('data/ForebrainGlut/hgForebrainGlut.loom', cleanup=True,
sparse=True, cache=True, backup_url='http://pklab.med.harvard.edu/velocyto/hgForebrainGlut/hgForebrainGlut.loom')

adata = scv.read("data/DentateGyrus/10X43_1.loom", sparse=True,
cache=True, backup_url="http://pklab.med.harvard.edu/velocyto/DG1/10X43_1.loom")
```
```
WARNING:root:initialising LoomConnection to data/ForebrainGlut/hgForebrainGlut.loom failed, closing file connection
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
  File "/path/to/local/python/python-dev/lib/python3.6/site-packages/scanpy/readwrite.py", line 78, in read
    backup_url=backup_url, cache=cache, **kwargs)
  File "/path/to/local/python/python-dev/lib/python3.6/site-packages/scanpy/readwrite.py", line 445, in _read
    adata = read_loom(filename=filename, **kwargs)
  File "/path/to/local/python/python-dev/lib/python3.6/site-packages/anndata/readwrite/read.py", line 152, in read_loom
    with connect(filename, 'r') as lc:
  File "/path/to/local/python/python-dev/lib/python3.6/site-packages/loompy/loompy.py", line 957, in connect
    return LoomConnection(filename, mode)
  File "/path/to/local/python/python-dev/lib/python3.6/site-packages/loompy/loompy.py", line 83, in __init__
    raise e
  File "/path/to/local/python/python-dev/lib/python3.6/site-packages/loompy/loompy.py", line 72, in __init__
    self.row_graphs = loompy.GraphManager(self, axis=0)
  File "/path/to/local/python/python-dev/lib/python3.6/site-packages/loompy/graph_manager.py", line 34, in __init__
    ds._file.create_group(a)
  File "/path/to/local/python/python-dev/lib/python3.6/site-packages/h5py/_hl/group.py", line 60, in create_group
    gid = h5g.create(self.id, name, lcpl=lcpl, gcpl=gcpl)
  File "h5py/_objects.pyx", line 54, in h5py._objects.with_phil.wrapper
  File "h5py/_objects.pyx", line 55, in h5py._objects.with_phil.wrapper
  File "h5py/h5g.pyx", line 161, in h5py.h5g.create
ValueError: Unable to create group (no write intent on file)
```

I tried to checkout to 0.1.10 and 0.1.6, same issue.
Are they new (missing) library requirements since v0.1.4?

Thanks.

**Comment by VolkerBergen** [comment]
First time seeing this error. Let's see how I can help. 

Could you send me your versions of all (in)direct dependencies using `scv.logging.print_versions()`?

**Comment by VolkerBergen** [comment]
@juugii Would appreciate to have your response on this so that I can help you with the issue. Thanks.

**Comment by juugii** [comment]
I am sorry for the late answer, I did answer by mail a few minutes after your post, but there is no trace of it here. I am just realizing you were waiting by logging on. Here is the versions I have been using:

scvelo==0.1.11.dev6+133abe9  scanpy==1.3.3+11.g623f0d4  anndata==0.6.13
numpy==1.14.6  scipy==1.1.0  matplotlib==3.0.0  sklearn==0.19.1
pandas==0.23.4

**Comment by juugii** [comment]
scvelo==0.1.13  scanpy==1.3.4  anndata==0.6.13  numpy==1.14.6  scipy==1.1.0  matplotlib==3.0.0  sklearn==0.19.1  pandas==0.23.4
I don't see the version checking function in the loompy complete API reference.

**Comment by juugii** [comment]
how can I send the loompy version after that scvelo update ?

**Comment by VolkerBergen** [comment]
Should be printed with `scv.logging.print_versions()`. If it is not shown, I'd assume that you don't have loompy installed. Just `pip install loompy`. You can check its version with 
```
import loompy
print(loompy.__version__)
```

**Comment by juugii** [comment]
I think we're getting closer of the source of my problem:
after pip install loompy:
```
Requirement already satisfied: loompy in /path/to/local/python/lib/python3.6/site-packages (2.0.2)
```
But:

```python
scv.logging.print_versions()
scvelo==0.1.13  scanpy==1.3.4  anndata==0.6.13  numpy==1.14.6  scipy==1.1.0  matplotlib==3.0.0  sklearn==0.19.1  pandas==0.23.4  
print(loompy.__version__)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: module 'loompy' has no attribute '__version__'
```

**Comment by VolkerBergen** [comment]
Yap, I guess we're getting closer. Upgrade with `pip install --upgrade loompy` and try again.

**Comment by juugii** [comment]
Now I see the source of the problem:
```python
pyscenic 0.8.9 has requirement llvmlite==0.22.0, but you'll have llvmlite 0.25.0 which is incompatible.
pyscenic 0.8.9 has requirement loompy==2.0.2, but you'll have loompy 2.0.16 which is incompatible.
pyscenic 0.8.9 has requirement numba==0.37.0, but you'll have numba 0.40.1 which is incompatible.
Installing collected packages: loompy
  Found existing installation: loompy 2.0.2
    Uninstalling loompy-2.0.2:
      Successfully uninstalled loompy-2.0.2
Successfully installed loompy-2.0.16
```
It is probably the right time to switch to venvs...

```python
import loompy
print(loompy.__version__)
2.0.16
```
And finaly
```python
import scvelo as scv
adata = scv.read('data/ForebrainGlut/hgForebrainGlut.loom', cleanup=True,
... sparse=True, cache=True, backup_url='http://pklab.med.harvard.edu/velocyto/hgForebrainGlut/hgForebrainGlut.loom')
... 100%Variable names are not unique. To make them unique, call `.var_names_make_unique`.
... storing 'Chromosome' as categorical
... storing 'Strand' as categorical
adata
AnnData object with n_obs × n_vars = 1720 × 32738 
    obs: 'Clusters'
    var: 'Accession', 'Chromosome', 'End', 'Start', 'Strand'
    layers: 'matrix', 'ambiguous', 'spliced', 'unspliced'
```
Problem solved!
Many thanks @VolkerBergen for your help, and sorry again for the delay in my answer.
Best.

**Comment by VolkerBergen** [comment]
Great that it works now!

**Comment by Mkebede** [comment]
I'm having another issue with reading a loom file that was generated from velocyto. The issue I'm getting is. 
```
ValueError: Row attribute 'Accession' dtype object is not allowed
Row attribute 'Chromosome' dtype object is not allowed
Row attribute 'Gene' dtype object is not allowed
Row attribute 'Strand' dtype object is not allowed
Column attribute 'CellID' dtype object is not allowed
For help, see http://linnarssonlab.org/loompy/format/
../../results/perturbseq_1/velocyto/perturbseq_1.loom does not appead to be a valid Loom file according to Loom spec version '2.0.1'
```

Running `scv.logging.print_versions()` produces:
```scvelo==0.1.23  scanpy==1.4.4.post1  anndata==0.6.22.post1  loompy==3.0.1  numpy==1.17.2  scipy==1.2.1  matplotlib==3.0.3  sklearn==0.21.3  pandas==0.25.1  ```

**Comment by VolkerBergen** [comment]
Try with `scv.read('filename' , validate=False)`.

**Comment by Mkebede** [comment]
Thank you for the suggestion! For me the issue was that the LoomValidator was outdated and incompatible with the latest version. Uninstalling and reinstalling the loom package seemed to address the issue. 

For anyone else having that issue, I did:
```
pip uninstall loom
pip install loom
```
Thank you again for your help!
